### PR TITLE
IPv6: fix deploy vm issue in ipv6-only networks without VR

### LINKED ISSUE #5640: NPE while deploy vm on a IPv6-only shared network
Got a NPE
```
java.lang.NullPointerException
        at com.cloud.network.Networks$BroadcastDomainType.getSchemeValue(Networks.java:177)
        at com.cloud.network.Networks$BroadcastDomainType.getValue(Networks.java:231)
        at com.cloud.hypervisor.kvm.resource.BridgeVifDriver.plug(BridgeVifDriver.java:209)
        at com.cloud.hypervisor.kvm.resource.LibvirtComputingResource.createVif(LibvirtComputingResource.java:3015)
        at com.cloud.hypervisor.kvm.resource.LibvirtComputingResource.createVifs(LibvirtComputingResource.java:2710)
        at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtStartCommandWrapper.execute(LibvirtStartCommandWrapper.java:81)
        at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtStartCommandWrapper.execute(LibvirtStartCommandWrapper.java:46)
        at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtRequestWrapper.execute(LibvirtRequestWrapper.java:78)
        at com.cloud.hypervisor.kvm.resource.LibvirtComputingResource.executeRequest(LibvirtComputingResource.java:1768)
```

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
management server
~~~

##### CLOUDSTACK VERSION

~~~
4.16.0.0
~~~

##### CONFIGURATION

advanced zone

##### OS / ENVIRONMENT

Mgmt server: Ubuntu 20.04
Hypervisor: Ubuntu 20.04

##### STEPS TO REPRODUCE

~~~
1. create a shared network with only ipv6
2. deploy a vm
~~~

##### EXPECTED RESULTS

~~~
succeed
~~~

##### ACTUAL RESULTS

~~~
Got exception, see the description
~~~

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR fixes #5640 partially.

Ipv6-only shared network with offering "Quickcloud with no services" works with this PR.
Ipv6-only shared network with other offerings, which require VR, still does not work, as VR cannot start.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [x] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test matrix

**Review by sureshanaparti — APPROVED** [review]
code LGTM

File: server/src/main/java/com/cloud/network/Ipv6AddressManagerImpl.java [hunk 1]
```diff
@@ -195,13 +195,23 @@ protected boolean isIp6Taken(Network network, String requestedIpv6) {
...
+                if (nic.getBroadcastType() == null) {
+                    nic.setBroadcastType(network.getBroadcastDomainType());
+                }
+                if (nic.getBroadCastUri() == null) {
+                    nic.setBroadcastUri(network.getBroadcastUri());
+                }
+                if (nic.getMacAddress() == null) {
+                    nic.setMacAddress(_networkModel.getNextAvailableMacAddressInNetwork(network.getId()));
+                }
+
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/network/Ipv6AddressManagerImpl.java:214]: I sugest a method `setNicPropertiesFromNetwork(NicProfile nic, Network network)`

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Review by GabrielBrascher — APPROVED** [review]
Thanks for the PR @weizhouapache.
+1 on Daan's suggestion on extracting those lines into a method.

Code LGTM.

File: server/src/main/java/com/cloud/network/Ipv6AddressManagerImpl.java (hunk 1 above)
Comment (weizhouapache) [reply server/src/main/java/com/cloud/network/Ipv6AddressManagerImpl.java]: done. thanks @sureshanaparti @DaanHoogland @GabrielBrascher for review.

**Review by GutoVeronezi — APPROVED** [review]

**Pull request merged**